Demos/Aspose.Email.Cloud.Live.Demos/Aspose.Email.Cloud.Live.Demos.UI/Scripts/Common/filedrop.js

```diff
@@ -0,0 +1,268 @@
+﻿// File Upload Component
+// options:
+// - DropFilesPrompt (string)
+// - ChooseFilePrompt (string)
+// - Accept (string)
+// - Multiple (boolean)
+// - UseSorting (boolean)
+// - MaximumUploadFiles (number)
+// - FileWrongTypeMessage (string)
+// - FileAmountMessage (string)
+// - FileSelectMessage (string)
+// - UploadOptions (array of string)
+// - UIBasePath (string)
+// - Method (callback)
+// - UploadAndRedirect (boolean)
+// - showAlert (function (string) => void)
+// - hideAlert (function () => void)
+// - showLoader (function () => void)
+// - progress (function () => void)
+
+(function ($) {
+    $.fn.filedrop = function (options) {
+        var getRandomIntInclusive = function (min, max) {
+            min = Math.ceil(min);
+            max = Math.floor(max);
+            return Math.floor(Math.random() * (max - min + 1)) + min;
+        };
+
+        var randomId = getRandomIntInclusive(1, Number.MAX_SAFE_INTEGER);
+        var acceptExts = options.Accept.split(/\s*,\s*/).map(function (ext) {
+            return ext.substring(1).toUpperCase();
+        });
+
+        var droppedFiles = [];
+
+        var getFileExtension = function (file) {
+            var pos = file.name.lastIndexOf('.');
+            return pos !== -1 ? file.name.substring(pos + 1).toUpperCase() : null;
+        };
+
+        var nextFileId = function () {
```

Comment:
This method looks strange. Looks like it always returns `2` if there is a file with the id `1`. Or runs forever instead.